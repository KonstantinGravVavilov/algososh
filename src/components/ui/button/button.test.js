import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe('testing Button component', () => {
  it('there is no button text', () => {
    const button = renderer
      .create(<Button />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it('there is button text', () => {
    const button = renderer
      .create(<Button text="test" />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it('button is disabled', () => {
    const button = renderer
      .create(<Button disabled />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it('button has loader', () => {
    const button = renderer
      .create(<Button isLoader />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it('button click', () => {
    const clickButton = jest.fn();

    render(<Button text="testButton" onClick={clickButton} />);
    fireEvent.click(screen.getByText("testButton"));
    expect(clickButton).toHaveBeenCalled();
  });

});
