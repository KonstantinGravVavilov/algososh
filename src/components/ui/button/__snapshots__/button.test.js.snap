// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button component button has loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`testing Button component button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`testing Button component there is button text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`testing Button component there is no button text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
